omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -274,10 +274,10 @@ public void shouldUpdateANonCodedDiagnosis() throws Exception {
 		
 		Assert.assertTrue(newDiagnosis.getVoided());
 		Assert.assertEquals(newDiagnosis.getDiagnosis().getCoded().getUuid(), concept.getUuid());
```

Comment:
```suggestion
		Assert.assertEquals(concept.getUuid(), newDiagnosis.getDiagnosis().getCoded().getUuid());
```
This is where we test newDiagnosis.getDiagnosis().getCoded().getUuid(), not concept.getUuid(). ;)